v2/tools/generator/internal/codegen/code_generator.go

```diff
@@ -154,7 +154,7 @@ func createAllPipelineStages(idFactory astmodel.IdentifierFactory, configuration
 		pipeline.AddSecrets(configuration).UsedFor(pipeline.ARMTarget),
 		pipeline.AddConfigMaps(configuration).UsedFor(pipeline.ARMTarget),
 
-		pipeline.CreateTypesForBackwardCompatibility("v2.0.0-alpha", configuration.ObjectModelConfiguration).UsedFor(pipeline.ARMTarget),
+		pipeline.CreateTypesForBackwardCompatibility("v2.0.0-", configuration.ObjectModelConfiguration).UsedFor(pipeline.ARMTarget),
```

Comment:
Shouldn't this be `v2.0.0-beta` - I think it's the prefix for the older versions.

Reply:
No because it needs to match anything that had a beta version, and we had a beta version for anything that had an alpha version, and anything that was added in beta.

So we need to match these two things:
`supportedFrom: v2.0.0-alpha.5` (added _before_ beta, so has a beta version)
`supportedFrom: v2.0.0-beta.0` (added _during_ beta, so has a beta version)

The only thing we don't want backcompat types for is `v2.0.0` (or `v2.1.0`, etc)

I've added a comment here to that effect for clarity.